provider/docker.go

```diff
@@ -50,16 +42,16 @@ func (provider *Docker) createClient() (client.APIClient, error) {
 		"User-Agent": "Traefik",
 	}
 	if provider.TLS != nil {
-		tlsOptions := tlsconfig.Options{
-			CAFile:             provider.TLS.CA,
-			CertFile:           provider.TLS.Cert,
-			KeyFile:            provider.TLS.Key,
-			InsecureSkipVerify: provider.TLS.InsecureSkipVerify,
-		}
-		config, err := tlsconfig.Client(tlsOptions)
+		config, err := provider.TLS.CreateTLSConfig()
 		if err != nil {
 			return nil, err
 		}
+		// TO DELETE IF USELESS : default docker TLS Client config
```

Comment:
According to my investigation, there is no need to setup `MaxVersion` and `CipherSuites`. 
I tried and succeeded  to connect Traefik to Docker via TLS with and without this section.
@emilevauge : Should I remove ?